tests/e2e/workloads/bdi/fixtures.py

```diff
@@ -0,0 +1,251 @@
+import pytest
+import os
+import logging
+from ocs_ci.utility.utils import run_cmd, clone_repo, exec_cmd
+from ocs_ci.ocs import machine, node, ocp, constants
+from ocs_ci.framework import config
+import tempfile
+from ocs_ci.utility import templating
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def clone_ibm_chart_project(request):
+    """
+    Clone IBM Chart project from Github
+    """
+    class_instance = request.node.cls
+
+    def finalizer():
+        """
+        Delete project
+        """
+        log.info(f"Deleting directory {class_instance.bdi_dir}")
+        if os.path.isdir(class_instance.bdi_dir):
+            run_cmd(cmd="rm -rf " + class_instance.bdi_dir)
+
+    request.addfinalizer(finalizer)
+
+    log.info(f"Cloning chart from github into {class_instance.bdi_dir}")
+    clone_repo(url=class_instance.chart_git_url, location=class_instance.bdi_dir)
+
+
+@pytest.fixture()
+def create_machineset(request):
+    """
+    Creates a machineset with replica 3.
+    Relevant only for AWS IPI deployment
```

Comment:
Will this test work with Non AWS IPI?

Reply:
Yes.
If we want to use Vsphere UPI, we need to create additional worker nodes during the post-deployment process.
Added it to the description